### DISCUSSION TITLE
Supabase triggers, functions & function hooks in production.

### DISCUSSION THREAD

**Discussion by Anan7Codes** [supabase/supabase#5392] [Questions]
Hey team, I've noticed that Supabase triggers, functions & function hooks are not production-ready, yet it's been shown a lot of tutorials out there. My app heavily depends on these features as well so can I go ahead and use them?!

**Comment by GaryAustin1** [comment]
Unofficial response:
The "not ready for production" warning is more about the UI for creating them.  Triggers and Functions are just native Postgres so are fine to use in production.  Function hooks are developed in this repository supabase/pg_net and had changes as recently as 19 days ago, so probably a bit riskier, probably though on the same level as most of Supabase.

**Reply by Anan7Codes** [reply]
Noted. Thanks for the reply man. Gonna be a little challenging if I can't use the hooks but I'm guessing it should be fine if I'm using HTTP requests to my own API right?

**Reply by GaryAustin1** [reply]
Certainly HTTP extension would work until they are "further along" or lose the alpha tag.   HTTP is sync though, so good to move from that as soon as possible if your external API would work with hooks  more limited features.

Edit: or if you meant use hooks anyway because you have a controlled environment, you probably should skim thru the pg_net repository issues and the like to see if you feel comfortable with what changes (if any) are going on.

**Reply by Anan7Codes** [reply]
Thanks Gary! This helps!

**Reply by ydennisy** [reply]
Thanks @GaryAustin1 - but it would be great to get a timeline from the supabase team @kiwicopple ?

**Reply by risonsimon** [reply]
That clears things up. Thanks. I was confused by the _not production ready_ label inside supabase UI. There should be a help text that clarifies that it's the UI that is not ready.
